File: Java/src/main/java/at/tugraz/experiments/DataReadIOSingle.java
package at.tugraz.experiments;

import at.tugraz.runtime.ObjectReader;
import at.tugraz.runtime.ObjectWriter;
import at.tugraz.util.Const;
import at.tugraz.util.RootData;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class DataReadIOSingle {

    public static void main(String[] args) throws IOException {

        String inDataPath = System.getProperty("inDataPath");
        String method = System.getProperty("method");
        String seqRand = System.getProperty("seqRand");
        int nrow = Integer.parseInt(System.getProperty("nrow"));

        ObjectReader reader = new ObjectReader(inDataPath, method);
        if (seqRand.equalsIgnoreCase("sequential")) {
            reader.readIO(0, nrow);
        }
        else {
            String randomDataPath = System.getProperty("randomDataPath");
            int[] randomIDs= new int[nrow];
            try (BufferedReader br = new BufferedReader(new FileReader(randomDataPath))) {
                String line;
                int index=0;
                while ((line = br.readLine()) != null) {
                    randomIDs[index++] = Integer.parseInt(line);
                }
                for (int i: randomIDs ) {
                    reader.readObjects(i, 1);
                }

            } catch (IOException e) {
                e.printStackTrace();
            }
        }

    }
}

File: src/main/rust/src/DataReadIOSingle.rs
#![allow(non_snake_case)]
#![allow(dead_code)]
#![allow(unused_variables)]

use std::{io, env};
use std::cmp::min;
use std::fs::File;
use std::io::{BufRead, BufReader};
use crate::runtime::ObjectReader::ObjectReader;
use crate::runtime::ObjectWriter::ObjectWriter;
use crate::tweetStructs::TweetStatus::TweetStatus;
use crate::util::Const::{BATCHSIZE, PAGESIZE};

mod tweetStructs;
mod runtime;
mod util;

fn main() -> io::Result<()> {
    let args: Vec<String> = env::args().collect();
    let inDataPath: &str = args[1].as_str();
    let method: &str = args[2].as_str();
    let seqRand: &str = args[3].as_str();
    let nrow: u32 = args[4].parse().unwrap();

    let mut reader = ObjectReader::new1(inDataPath, method);
    if seqRand.to_lowercase().eq("sequential") {
        reader.readIO(0, nrow);
    }
    else {
        let randomDataPath: &str = args[5].as_str();
        let infile = File::open(randomDataPath)?;
        let f = BufReader::new(infile);
        for line in f.lines() {
            reader.readObjectIO(line.unwrap().parse().unwrap());
        }
    }
    reader.flush();
    Ok(())
}